src/main/java/org/jabref/model/pdf/search/SearchResult.java

```diff
@@ -0,0 +1,89 @@
+package org.jabref.model.pdf.search;
+
+import java.io.IOException;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.LinkedFile;
+
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.index.IndexableField;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.highlight.Highlighter;
+import org.apache.lucene.search.highlight.InvalidTokenOffsetsException;
+import org.apache.lucene.search.highlight.QueryScorer;
+import org.apache.lucene.search.highlight.SimpleHTMLFormatter;
+import org.apache.lucene.search.highlight.TextFragment;
+import org.apache.lucene.search.highlight.TokenSources;
+
+import static org.jabref.model.pdf.search.SearchFieldConstants.CONTENT;
+import static org.jabref.model.pdf.search.SearchFieldConstants.MODIFIED;
+import static org.jabref.model.pdf.search.SearchFieldConstants.PATH;
+
+public final class SearchResult {
+
+    private final String path;
+    private final String content;
+    private final long modified;
+
+    private final float luceneScore;
+    private String html;
+
+    public SearchResult(IndexSearcher searcher, Query query, ScoreDoc scoreDoc) throws IOException {
+        this.path = getFieldContents(searcher, scoreDoc, PATH);
+        this.content = getFieldContents(searcher, scoreDoc, CONTENT);
+        this.modified = Long.valueOf(getFieldContents(searcher, scoreDoc, MODIFIED));
+        this.luceneScore = scoreDoc.score;
+
+        TokenStream stream = TokenSources.getTokenStream(CONTENT, content, new EnglishStemAnalyzer());
```

Comment:
The whole tokenization of Lucene seems to be reworked and TokenSources.getTokenStream is marked deprecated. In the lates api documentation of lucene this doesn't even appear anymore. So this needs somehow to be redone.
Maybe you'll find some resources here (https://lucene.apache.org/core/8_9_0/core/org/apache/lucene/analysis/package-summary.html) or some hints here (https://stackoverflow.com/a/50785820).